react/components/TopBar/components/PrimarySearch/components/PrimarySearchResults/index.js

```diff
@@ -0,0 +1,96 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Query } from 'react-apollo';
+
+import mod from 'react/util/mod';
+
+import primarySearchResultsQuery from 'react/components/TopBar/components/PrimarySearch/components/PrimarySearchResults/queries/primarySearchResults';
+
+import Text from 'react/components/UI/Text';
+import PrimarySearchResult from 'react/components/TopBar/components/PrimarySearch/components/PrimarySearchResults/PrimarySearchResult';
+
+export default class PrimarySearchResults extends PureComponent {
+  static propTypes = {
+    query: PropTypes.string.isRequired,
+    cursor: PropTypes.number,
+    onSelection: PropTypes.func,
+  }
+
+  static defaultProps = {
+    cursor: null,
+    onSelection: () => {},
+  }
+
+  selectResult = (result) => {
+    const { query, onSelection } = this.props;
+
+    if (result) return onSelection(result.href);
+
+    return onSelection(`/search/${encodeURIComponent(query)}`);
+  }
+
+  render() {
+    const { query, cursor } = this.props;
+
+    return (
+      <Query query={primarySearchResultsQuery} variables={{ query }}>
+        {({ data, loading, error }) => {
+          const { searches } = data;
+
+          if (loading && !searches) {
+            this.selectResult();
+
+            return (
+              <PrimarySearchResult>
+                <Text fontWeight="bold">
+                  Searching...
+                </Text>
+              </PrimarySearchResult>
+            );
+          }
+
+          if (error) {
+            this.selectResult();
+
+            return (
+              <PrimarySearchResult>
+                <Text fontWeight="bold" color="state.alert">
+                  {error.message}
+                </Text>
+              </PrimarySearchResult>
+            );
+          }
+
+          const { results } = searches;
+          const selected = cursor && mod(cursor, results.length + 1);
+
+          this.selectResult(results[selected]);
+
+          return (
+            <React.Fragment>
```

Comment:
Interestingggg

Reply:
what's interesting?

Reply:
Have not seen this before

Reply:
Ah, it's just so you can return a single child without an extra DOM element, like wrapping something in a <div>